### PR TITLE
Better notifications

### PR DESCRIPTION
**Pull request by cybrox**
- Will mark notifications as read when they are clicked

cc @vevix

### CODE REVIEW

File: app/mixins/get-owner.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import Ember from 'ember';
+import set from 'ember-metal/set';
+
+const { getOwner } = Ember;
+
+/**
+ * Simple mixing for using getOwner
```
Comment (cybrox) [inline app/mixins/get-owner.js]: _"mixing"_, yep, I need more sleep 💣

File: app/components/application/user-notifications/item.js [hunk 2]
```diff
@@ -56,14 +55,20 @@ export default Component.extend({
...
+          return null;
         }
       }
     }
-  })
+  }),
+
+  actions: {
+    followNotification() {
+      this.sendAction('followNotification', get(this, 'group'), get(this, 'target'));
```
Comment (trmcnvn) [inline app/components/application/user-notifications/item.js]: `invokeAction(this, ...)`

File: app/mixins/get-owner.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import Ember from 'ember';
+import set from 'ember-metal/set';
+
+const { getOwner } = Ember;
```
Comment (trmcnvn) [inline app/mixins/get-owner.js]: `import ... from ember-owner/get`

File: app/components/stream-feed/items/post/comments.js [hunk 4]
```diff
@@ -60,6 +60,22 @@ export default Component.extend({
...
+    if (get(attrs, 'oldAttrs.forceUpdate.value') === get(attrs, 'newAttrs.forceUpdate.value')) {
+      return;
+    }
+
+    set(this, 'comments', []);
+    get(this, 'getComments').perform().then((comments) => {
+      const content = comments.toArray().reverse();
+      set(content, 'links', get(comments, 'links'));
+      set(this, 'comments', content);
+    }).catch(() => {});
```
Comment (trmcnvn) [inline app/components/stream-feed/items/post/comments.js:60]: Logic should be moved to a function or the task itself as it's now duplicated.

File: app/mixins/get-owner.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+/**
+ * Simple mixing for using getOwner
+ *
+ * getOwner is the ember 2.x public api version of ember 1.x old __container__ lookups.
+ * This mixin will simply add the getOwner method to an object to allow global lookups.
+ * Reference: emberjs.com/deprecations/v2.x/#toc_id-ember-application-injected-container
+ */
+export default Ember.Mixin.create({
```
Comment (trmcnvn) [inline app/mixins/get-owner.js]: `import ... from 'ember-metal/mixin'`

**Review by trmcnvn — CHANGES_REQUESTED** [review]
This approach has a few issues.

First, if the comment referenced in the notification is past the fold (not the 2 most recent), then reloading the comments isn't going to show that content anyway.

Second, if it's a brand new comment then implementing realtime of comments would negate the need for this.

------

I think the better approach here is to drop the hacks and keep the mark as read code. Eventually direct comment linking (via hash or query param) & realtime will fix both issues here.

**Comment by cybrox** [comment]
Dropped the hacks, mark as read, then transition with injected router.

**Review by trmcnvn — APPROVED** [review]

**Pull request merged**